### PR TITLE
Issue 89: UI to edit non-ui driven xp fields

### PR DESCRIPTION
**Pull request by djsteinmetz**
- Add UI to create/edit/delete non ui related xp fields

### CODE REVIEW

File: src/components/products/detail/ProductDetail.tsx [hunk 1]
```diff
@@ -204,13 +213,104 @@ export default function ProductDetail({
...
+        <CardBody>
+          <Box
+            p={6}
+            display="flex"
+            flexDirection={"column"}
+            alignItems={"flex-start"}
+            justifyContent={"center"}
+            minH={"xs"}
+          >
+            {Object.values(nonUiXp)?.map((xp, idx) => {
```
Comment (crhistianramirez) [inline src/components/products/detail/ProductDetail.tsx:255]: nitpick: can we remove `?.` here, not needed

File: src/components/products/detail/ProductDetail.tsx [hunk 2]
```diff
@@ -204,13 +213,104 @@ export default function ProductDetail({
...
+              return (
+                <Box key={idx} my={1} display="flex" justifyContent="flex-start" alignItems="center">
+                  <Button
+                    variant="outline"
+                    size="sm"
+                    colorScheme="primary"
+                    onClick={() => {
+                      setXpPropertyNameToEdit(Object.keys(nonUiXp)?.[idx])
+                      setXpPropertyValueToEdit(xp)
```
Comment (crhistianramirez) [inline src/components/products/detail/ProductDetail.tsx:264]: nitpick: can we remove `?.` here, not needed

File: src/components/products/detail/ProductDetail.tsx [hunk 3]
```diff
@@ -204,13 +213,104 @@ export default function ProductDetail({
...
+                  </Button>
+                  <Text
+                    minWidth="7rem"
+                    fontSize="0.8rem"
+                    fontWeight="bold"
+                    color="blackAlpha.400"
+                    textTransform="uppercase"
+                    letterSpacing={1}
+                  >
+                    {Object.keys(nonUiXp)?.[idx]}
```
Comment (crhistianramirez) [inline src/components/products/detail/ProductDetail.tsx:288]: nitpick: can we remove `?.` here, not needed

**Review by crhistianramirez — CHANGES_REQUESTED** [review]
Can you clean up all instances of the `XpCard` component (old xp builder)

**Comment by djsteinmetz** [comment]
All comments have been addressed. I also created a new issue related, to use this genericize this xp component and use it for all resources. #144

**Review by crhistianramirez — APPROVED** [review]

**Comment by robertsoniv** [comment]
@djsteinmetz @crhistianramirez - I realize it probably worked this way before, but do we think that it makes sense to track XP changes the same as other tabs? Where it only saves changes when the user hits the Save button and the "discard changes" will undo anything you changed in Xp tab?

Thinking like if you delete or change a property then want to "discard" those changes. Also less saves / syncs if we save all changes at once - so in theory more performant.

**Review by robertsoniv — APPROVED** [review]

**Pull request merged**